core/src/main/scala/kafka/server/KafkaRequestHandler.scala

```diff
@@ -141,10 +142,10 @@ class KafkaRequestHandlerPool(val brokerId: Int,
   }
 }
 
-class BrokerTopicMetrics(name: Option[String]) extends KafkaMetricsGroup {
+class BrokerTopicMetrics(name: Option[TopicPartition]) extends KafkaMetricsGroup {
   val tags: scala.collection.Map[String, String] = name match {
     case None => Map.empty
-    case Some(topic) => Map("topic" -> topic)
+    case Some(topic) => Map("topic" -> topic.topic(), "partition" -> topic.partition().toString)
```

Comment:
Note that this causes an explosion in the number of metrics and it will break JMX users. As such, a KIP would be required.

Reply:
@ijuma Got it